enclave_manager/avalon_enclave_manager/base_enclave_manager.py

```diff
@@ -0,0 +1,287 @@
+        started and before it handles requests indefinitely. It may include
+        a cleanup of database or ensuring syncup with it. There can also be
+        some sanity checks to ensure reentrancy in case of abrupt failures.
+        """
+        pass
+
+# -------------------------------------------------------------------------
+
+    @abstractmethod
+    def _execute_work_order(self, input_json_str, indent=4):
+        """
+        Submits the incoming request to corresponding enclave for execution.
+
+        Returns :
+            response - Response received as response from execution at the
+                       enclave.
+        """
+        pass
+
+# -------------------------------------------------------------------------
+
+    @abstractmethod
+    def start_enclave_manager(self):
+        """
+        Executes boot flow and run time flow. This is the engine of an enclave
+        manager which keeps running as long as there are no unexpected errors.
+        It should first carry out the boot time flow followed by the run time
+        routine which is indefinite.
+        """
+        pass
+
+# -------------------------------------------------------------------------
+
+    def _connect_to_kv_store(self):
+        """
+        This function creates a connection to the KVStorage.
+
+        Returns :
+            kv_helper - An instance of LMDBHelperProxy that helps interact
+                        with the database
+        """
+        if self._config.get("KvStorage") is None:
+            logger.error("Kv Storage path is missing")
+            sys.exit(-1)
+        try:
+            kv_helper = connector.open(self._config['KvStorage']['remote_url'])
+        except Exception as err:
+            logger.error("Failed to open KV storage interface; " +
+                         "exiting Intel SGX Enclave manager: {}".format(err))
+            sys.exit(-1)
+        return kv_helper
+
+# -------------------------------------------------------------------------
+
+    def _setup_enclave(self):
+        """
+        This function is responsile for initialization of the trusted
+        enclave. It does so via the EnclaveInfo class which not only
+        initializes the enclave but also ensure the enclave creates
+        the signup data. This signup data is key to all the trusted
+        functionalities over this instance of the enclave. Therefore
+        all relevant data is sent back here. This is kept in memory
+        for use hereafter.
+
+        Returns :
+            signup_data - Relevant signup data to be used for requests to the
+                          enclave
+            extended_measurements - A tuple of enclave basename & measurements
+        """
+        try:
+            logger.info("Initialize enclave and create signup data")
+            signup_data = enclave_info.\
+                EnclaveInfo(self._config.get("EnclaveModule"))
+            # Extended measurements is a list of enclave basename and
```

Comment:
Can this comment be removed as you already covered this in the function description?